### DISCUSSION TITLE
[done] PHP Style Guide

### DISCUSSION THREAD

**Discussion by Thijmen** [coollabsio/coolify#2425] [Feature Requests]
Hi there @andrasbacsai,

Seeing as the project gains more traction, would it be a good idea to have PHP Style guide being enforced throughout GH Actions?

Would be nice to have 1 uniform way to have PHP classes/code in the files.

Let me know if I can help out here, I am more than happy to create a PR which adds these checks to PRs.

**Comment by andrasbacsai** [comment]
Hey, that sounds very good! Can you please open a PR or let me know what do you need from my side? 🙂

**Reply by Thijmen** [reply]
Hey, thanks for getting back with me! Sure, I'll draft something up.

I will try to come up with some basic set of rules which most publishers use. In my case, I like to take a look at how Spatie does things (for example, https://github.com/spatie/laravel-activitylog/blob/main/.php_cs.laravel).

I can imagine that once I open that PR, some actions will fail because the code is not adhering on all places to the style we set. I will therefore also fix that code style to make it compliant.

How do you feel about my proposal?

**Reply by Thijmen** [reply]
I've created https://github.com/coollabsio/coolify/pull/2438! :) Keen on hearing your opinion.

**Reply by andrasbacsai** [reply]
> Hey, thanks for getting back with me! Sure, I'll draft something up.
> 
> I will try to come up with some basic set of rules which most publishers use. In my case, I like to take a look at how Spatie does things (for example, https://github.com/spatie/laravel-activitylog/blob/main/.php_cs.laravel).
> 
> I can imagine that once I open that PR, some actions will fail because the code is not adhering on all places to the style we set. I will therefore also fix that code style to make it compliant.
> 
> How do you feel about my proposal?

I love this. I already merged the PR, but we need to make a `pint.json` with the right rules, but I have literally 0 idea how to configure it.

I guess you have more experience with Pint than me (0). 😅

**Reply by Thijmen** [reply]
> > Hey, thanks for getting back with me! Sure, I'll draft something up.
> > I will try to come up with some basic set of rules which most publishers use. In my case, I like to take a look at how Spatie does things (for example, https://github.com/spatie/laravel-activitylog/blob/main/.php_cs.laravel).
> > I can imagine that once I open that PR, some actions will fail because the code is not adhering on all places to the style we set. I will therefore also fix that code style to make it compliant.
> > How do you feel about my proposal?
> 
> I love this. I already merged the PR, but we need to make a `pint.json` with the right rules, but I have literally 0 idea how to configure it.
> 
> I guess you have more experience with Pint than me (0). 😅

What rules are you thinking about? Let me create a draft PR.

**Reply by Thijmen** [reply]
@andrasbacsai I've send you a message on Discord, we could chat from there too.
